plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2519,10 +2517,12 @@ private Optional<SystemTable> getRawSystemTable(ConnectorSession session, Schema
 
     private static Map<String, DeltaLakeColumnStatistics> toDeltaLakeColumnStatistics(Collection<ComputedStatistics> computedStatistics)
     {
-        // Only statistics for whole table are collected
-        ComputedStatistics singleStatistics = Iterables.getOnlyElement(computedStatistics);
-        return createColumnToComputedStatisticsMap(singleStatistics.getColumnStatistics()).entrySet().stream()
-                .collect(toImmutableMap(Entry::getKey, entry -> createDeltaLakeColumnStatistics(entry.getValue())));
+        return computedStatistics.stream()
```

Comment:
`.filter(statistics -> statistics.getGroupingColumns().isEmpty())` ?

Reply:
i see i am wrong; this MAY be collected on per-file basis (per next commit)

(i am not convinced the commit extraction really helped)